src/ui/osx/TogglDesktop/Feature/TimeEntry/TimeEntryDatasource.swift

```diff
@@ -272,6 +283,11 @@ extension TimeEntryDatasource {
                                 forSupplementaryViewOfKind:NSCollectionView.elementKindSectionHeader,
                                 withIdentifier: Constants.TimeHeaderID)
     }
+
+    fileprivate func lastDateInSections() -> Date? {
+        let lastSection = sections.reversed().first { !$0.isLoadMore }
```

Comment:
can you use `last` instead of `reversed().first`?

Reply:
Actually, I deliberately choose `reversed().first {where: {}}` since it only costs two if-comparison steps at maximum. 

Additionally, The O(n) of `reversed()` is 1, since it doesn't reverse entire the array. it's just a reversed iterator.
Ref: https://developer.apple.com/documentation/swift/array/1690025-reversed

---------
If we use `last {where: {}}`, it will iterates from the beginning and find the item at end of the list => O(n) = n, which is unnecessary.